### PR TITLE
Dep spike

### PR DESCRIPTION
**Pull request by ebrygg**
The Chain Saw Massacre, have a look.

### CODE REVIEW

File: application/pom.xml [hunk 1]
```diff
@@ -124,10 +124,10 @@
             <artifactId>asset-client</artifactId>
         </dependency>
 
-        <dependency>
+        <!--dependency>
```
Comment (RobinMilding) [inline application/pom.xml]: We shouldn't leave commented out code

File: application/pom.xml [hunk 2]
```diff
@@ -139,6 +139,10 @@
             <artifactId>uvms-config</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>eu.europa.ec.fisheries.uvms.config</groupId>
+            <artifactId>config-model</artifactId>
```
Comment (RobinMilding) [inline application/pom.xml:148]: Why was this needed now?

File: application/src/main/java/eu/europa/ec/fisheries/uvms/activity/rest/resources/ReportDocumentResource.java [hunk 3]
```diff
@@ -43,10 +45,23 @@ This file is part of the Integrated Fisheries Data Management (IFDM) Suite. The
     @Produces(value = {MediaType.APPLICATION_XML})
     @Consumes(value = {MediaType.APPLICATION_XML})
     public Object getReport(FLUXFAQueryMessage fluxfaQueryMessage) throws JAXBException {
-        SubscriptionDataRequest subscriptionDataRequest = SubscriptionMapper.mapToSubscriptionDataRequest(fluxfaQueryMessage.getFAQuery());
-        List<SubscriptionDataCriteria> queryCriteria = subscriptionDataRequest.getQuery().getCriteria();
-        log.debug(queryCriteria.toString());
-        FLUXFAReportMessage reportsByCriteria = faQueryService.getReportsByCriteria(queryCriteria);
+        DateTimeType startDate = fluxfaQueryMessage.getFAQuery().getSpecifiedDelimitedPeriod().getStartDateTime();
```
Comment (RobinMilding) [inline application/src/main/java/eu/europa/ec/fisheries/uvms/activity/rest/resources/ReportDocumentResource.java]: Maybe some input checks so we can return a nice error message if the query is missing or has no specifiedDelimitedPeriod?

File: application/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/bean/FAQueryServiceBean.java [hunk 4]
```diff
@@ -37,26 +34,8 @@ public void init() {
...
-                    startDate = Instant.parse(value);
-                }
-            }
-
-            List<FaReportDocumentEntity> faReportDocumentsForTrip = faReportDao.loadReports(null, "N", startDate, endDate);
-            return ActivityEntityToModelMapper.INSTANCE.mapToFLUXFAReportMessage(faReportDocumentsForTrip);
-        }
-        return null;
+    public FLUXFAReportMessage getReportsByCriteria(Instant startDate, Instant endDate) {
+        List<FaReportDocumentEntity> faReportDocumentsForTrip = faReportDao.loadReports(null, "N", startDate, endDate);
```
Comment (RobinMilding) [inline application/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/bean/FAQueryServiceBean.java:38]: This is quite a bit nicer!

File: client-model/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/search/FishingActivityQuery.java [hunk 5]
```diff
@@ -15,34 +15,26 @@ This file is part of the Integrated Fisheries Data Management (IFDM) Suite. The
...
 public class FishingActivityQuery {
 
-    private PaginationDto pagination;
     private Map<SearchFilter, String> searchCriteriaMap;
     private Map<SearchFilter, List<String>> searchCriteriaMapMultipleValues;
     private SortKey sorting;
     private List<GroupCriteria> groupByFields;
     private Boolean showOnlyLatest;
     private Boolean useStatusInsteadOfPurposeCode;
+    private Integer offset;
```
Comment (RobinMilding) [inline client-model/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/search/FishingActivityQuery.java:31]: This change could have been a separate commit

File: application/pom.xml [hunk 6]
```diff
@@ -157,11 +158,6 @@
             <artifactId>lombok</artifactId>
         </dependency>
 
-        <dependency>
```
Comment (RobinMilding) [inline application/pom.xml:155]: Good cleanup but the commit message could have had a bit more explanation of what happened :)

File: application/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/util/GeomUtil.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        Coordinate coordinate = lengthIndexedLine.extractPoint(index);
+        return createPoint(coordinate);
+    }
+
+    public static LengthIndexedLine createLengthIndexedLine(String wkt1, String wkt2) throws Exception {
+        Geometry lineString = GeomUtil.createLineString(wkt1, wkt2);
+        return new LengthIndexedLine(lineString);
+    }
+
+    public static Geometry createLineString(String wkt1, String wkt2) throws Exception {
```
Comment (RobinMilding) [inline application/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/util/GeomUtil.java]: Can this throw so many types of exceptions that it warrants "throw Exception"?

File: application/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FeatureToGeoJsonJacksonMapper.java [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+                        arrayNode.add(o.toString());
+                    }
+                    obj.putArray(property.getName().toString()).addAll(arrayNode);
+                }else if(List.class.equals(property.getType().getBinding())) {
+                    ArrayNode arrayNode = mapper.createArrayNode();
+                    for (Object o : (ArrayList) value) {
+                        arrayNode.add(o.toString());
+                    }
+                    obj.putArray(property.getName().toString()).addAll(arrayNode);
+                }
```
Comment (RobinMilding) [inline application/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FeatureToGeoJsonJacksonMapper.java:95]: The code in the if- and else if-clause are exactly the same

File: application/src/test/java/eu/europa/ec/fisheries/uvms/activity/rest/BaseActivityArquillianTest.java [hunk 9]
```diff
@@ -136,7 +135,7 @@ protected void setUp() throws NamingException, ServiceException, JAXBException,
     }
 
     @Transactional
-    private void populateFluxTestData() throws IOException, JAXBException, ServiceException {
+    public void populateFluxTestData() throws IOException, JAXBException, ServiceException {
```
Comment (RobinMilding) [inline application/src/test/java/eu/europa/ec/fisheries/uvms/activity/rest/BaseActivityArquillianTest.java:138]: Why is this now public?

File: application/src/test/java/eu/europa/ec/fisheries/uvms/activity/service/bean/FluxMessageServiceBeanTest.java [hunk 10]
```diff
@@ -80,18 +77,16 @@ public void setUp() {
         fluxFaReportMessage.setFLUXReportDocument(MapperUtil.getFluxReportDocumentCorrection());
 
         Mockito.doReturn(getMockedFishingActivityReportEntity()).when(faReportDocumentDao).findFaReportByIdAndScheme(any(String.class), any(String.class));
-        Mockito.when(fluxFaReportMessageDao.createEntity(any(FluxFaReportMessageEntity.class))).then(invocation -> invocation.getArguments()[0]);
     }
 
     @Test
+    @Ignore
```
Comment (RobinMilding) [inline application/src/test/java/eu/europa/ec/fisheries/uvms/activity/service/bean/FluxMessageServiceBeanTest.java]: A short description of why the tests are ignored would be nice (even if  the test is crap)

File: application/src/main/java/eu/europa/ec/fisheries/uvms/activity/rest/resources/ReportDocumentResource.java [hunk 11]
```diff
@@ -45,23 +45,9 @@ This file is part of the Integrated Fisheries Data Management (IFDM) Suite. The
...
-                startDate.getDateTime().getSecond());
-        LocalDateTime localEndDate = LocalDateTime.of(
-                endDate.getDateTime().getYear(),
-                endDate.getDateTime().getMonth(),
-                endDate.getDateTime().getDay(),
-                endDate.getDateTime().getHour(),
-                endDate.getDateTime().getMinute(),
-                endDate.getDateTime().getSecond());
-        FLUXFAReportMessage reportsByCriteria = faQueryService.getReportsByCriteria(localStartDate.toInstant(ZoneOffset.UTC), localEndDate.toInstant(ZoneOffset.UTC));
+        Instant startInstant = fluxfaQueryMessage.getFAQuery().getSpecifiedDelimitedPeriod().getStartDateTime().getDateTime().toGregorianCalendar().getTime().toInstant();
```
Comment (RobinMilding) [inline application/src/main/java/eu/europa/ec/fisheries/uvms/activity/rest/resources/ReportDocumentResource.java:48]: Prettier but not pretty, but what can you do :)
I'm a bit worried though that the UTC ZoneOffset disappeared.

File: application/pom.xml (hunk 1 above)
Comment (ebrygg) [reply application/pom.xml]: Already fixed?

File: application/pom.xml (hunk 2 above)
Comment (ebrygg) [reply application/pom.xml:148]: Later version of uvms-config had removed it from the dependency tree

File: application/src/main/java/eu/europa/ec/fisheries/uvms/activity/rest/resources/ReportDocumentResource.java (hunk 3 above)
Comment (ebrygg) [reply application/src/main/java/eu/europa/ec/fisheries/uvms/activity/rest/resources/ReportDocumentResource.java]: That is already handled by the API, if none is specified then search retrieves everything

File: application/src/test/java/eu/europa/ec/fisheries/uvms/activity/rest/BaseActivityArquillianTest.java (hunk 9 above)
Comment (ebrygg) [reply application/src/test/java/eu/europa/ec/fisheries/uvms/activity/rest/BaseActivityArquillianTest.java:138]: Interceptors do not work for private methods

File: application/src/test/java/eu/europa/ec/fisheries/uvms/activity/service/bean/FluxMessageServiceBeanTest.java (hunk 10 above)
Comment (ebrygg) [reply application/src/test/java/eu/europa/ec/fisheries/uvms/activity/service/bean/FluxMessageServiceBeanTest.java]: Well, next step is to remove these tests completly

File: application/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FeatureToGeoJsonJacksonMapper.java (hunk 8 above)
Comment (ebrygg) [reply application/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FeatureToGeoJsonJacksonMapper.java:95]: Yes, this file is copied from commons-geometry, there might be dragons here, as with any code from the commons dependencies

File: application/src/test/java/eu/europa/ec/fisheries/uvms/activity/service/bean/FluxMessageServiceBeanTest.java (hunk 10 above)
Comment (ebrygg) [reply application/src/test/java/eu/europa/ec/fisheries/uvms/activity/service/bean/FluxMessageServiceBeanTest.java]: Tests removed

File: application/src/main/java/eu/europa/ec/fisheries/uvms/activity/rest/resources/ReportDocumentResource.java (hunk 11 above)
Comment (ebrygg) [reply application/src/main/java/eu/europa/ec/fisheries/uvms/activity/rest/resources/ReportDocumentResource.java:48]: Don't think that is it a problem in this case.

**Review by RobinMilding — APPROVED** [review]

**Review by robertkrantz — APPROVED** [review]

**Review by psjobergprivat — COMMENTED** [review]
Let the chainsaw rule!
